seed/java-sdk/pagination/src/main/java/com/seed/pagination/core/pagination/SyncPagingIterable.java

```diff
@@ -0,0 +1,52 @@
+package com.hume.api.core.pagination;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+public class SyncPagingIterable<T> extends SyncPage<T> implements Iterable<T> {
+
+    public SyncPagingIterable(boolean hasNext, List<T> items, Supplier<? extends SyncPage<T>> getNext) {
+        super(hasNext, items, getNext);
+    }
+
+    public SyncPagingIterable(boolean hasNext, Optional<List<T>> items, Supplier<? extends SyncPage<T>> getNext) {
+        super(hasNext, items.orElse(new ArrayList<>()), getNext);
+    }
+
```

Comment:
In addition to the itererator, should we have a way to get all of the values in a list or stream easily? like a `toList()`?

my vote: make a ticket and consider adding this in the future